src/rust/engine/process_execution/src/local.rs

```diff
@@ -368,12 +372,17 @@ pub trait CapturedWorkdir {
       //   https://github.com/pantsbuild/pants/issues/6089
       .map(ChildResults::collect_from)
       .and_then(move |child_results_future| {
-        timeout(req_timeout, child_results_future)
-          .boxed()
-          .compat()
-          .map_err(|e| e.to_string())
+        let maybe_timed_out_child_results = Box::pin(async move {
```

Comment:
Yay async await! I wrestled for a long time to get local timeouts working.

Reply:
I'm planning on porting the most relevant bits of the process_execution crate before following up on the remote portion... too fiddly otherwise, heh.